pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -101,6 +110,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 			return reconcile.Result{}, err
 		}
 
+		if mhc, ok := resource.(*machinev1beta1.MachineHealthCheck); ok {
+			isUpgrading, err := r.isClusterUpgrading(ctx)
+			if err != nil {
+				r.Log.Error(err)
+				r.SetDegraded(ctx, err)
+
+				return reconcile.Result{}, err
+			}
+
+			if isUpgrading {
+				mhc.ObjectMeta.Annotations = map[string]string{
+					MHCPausedAnnotation: "",
+				}
+				shouldRequeue = true
```

Comment:
instead of using a boolean here, can we declare `requeueAfter` on line 100 and then set it to 1hr here?

Unless we had other plans for `shouldRequeue` I don't see any other use for it now.

Reply:
I can make that change. I also think we can just remove the requeueing entirely, since the clusterversion resource will update itself during an upgrade, causing the reconciler here to fire again. I can test this.

Reply:
I've confirmed that requeueing is completely unnecessary for cluster upgrades since the updated clusterversion resource will invoke a reconcile anyways. 

I'll push up a change to remove it entirely after I get some better test coverage around it.